test/common/extensions/retry/options/network_configuration/predicate_test.cc

```diff
@@ -0,0 +1,46 @@
+#include "envoy/registry/registry.h"
+#include "envoy/upstream/retry.h"
+
+#include "source/common/upstream/retry_factory.h"
+
+#include "test/mocks/server/factory_context.h"
+#include "test/mocks/stream_info/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "library/common/extensions/retry/options/network_configuration/config.h"
+#include "library/common/network/configurator.h"
+
+using namespace testing;
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Options {
+namespace {
+
+TEST(NetworkConfigurationRetryOptionsPredicateTest, PredicateTest) {
```

Comment:
Initial test that confirms proper behavior of:
- factory registration.
- death on predicate creation without a configurator.
- correct predicate creation.